lips/lip-10.md

```diff
@@ -0,0 +1,234 @@
+* For the merchant use case, Libra ID allows users to self-identify to merchants. In this case, Libra ID plays the role of credit card number or also can be seen as a variant of OpenID-type protocol.
+
+## Examples in other blockchains
+Ripple proposes PayID. PayID has some similarity to the current proposal, but is not entirely identical.
+For example, PayID seems to target cross-chain payments, while libra id is intended to facilitate cross-vasp payments, but only inside the libra network.
+
+# Libra ID format
+Libra ID is a string in format `<user-id>$<libra-id-domain>`.
+
+* **user-id**, identifies an account within a VASP:
+  * Case insensitive
+  * Regex: `^[a-zA-Z0-9][a-zA-Z0-9-_+.]?$`
+  * Maximum length: 64 characters
+  * Not published on the chain, there is no way for an external entity to list all the users within VASP.
+* **libra-id-domain** identifies the VASP:
+  * Case insensitive
+  * Regex: `^[a-zA-Z0-9][a-zA-Z0-9-]?$`
+  * Maximum length: 63 characters
+  * libra-id-domains are published on the chain
+  * Each libra-id-domain is uniquely associated with single VASP
+  * VASPs can have multiple libra-id-domains associated, though it is not expected to be common
+
+## Multiple accounts per user
+VASPs should not assume that Libra ID uniquely identifies the user within the system. Users that have accounts with different VASPs would have different Libra ID within each VASP.
+
+Furthermore, a user can have multiple Libra IDs within the same VASP. This does not affect the protocol, but VASPs can leverage this - for example to allow users to create multiple ‘accounts’ like checking, savings, etc.
+
+VASPs can also allow users to generate temporary / situational Libra IDs. This can be used in many scenarios such as:
+
+* Temporary Libra ID for some merchant payments, when user don’t want to share their ‘main’ Libra ID with the merchant
+* Publicly shared Libra IDs for the various campaigns, like donations, etc.
+
+## Libra ID domains
+Each libra ID domain is associated with a VASP. The source of truth for this association is a blockchain - a special LibraIdDomains resource must be published into VASP for it to expose it’s LibraID domain.
+
+```
+resource struct LibraIdDomains {
+    domains: vector<LibraIdDomain>,
+}
+
+struct LibraIdDomain {
+    domain: vector<u8>,     // Utf-8 encoding
+    wallet_uri: vector<u8>, // ASCII encoding (use url-escape for non-ASCII characters)
+}
+```
+
+* The `LibraIdDomains` resource can only be published into an account with `Role == PARENT_VASP_ROLE_ID`.
+* The `LibraIdDomains` contains a list of `LibraIdDomain` structs that can be associated with a VASP. As such, it is possible to register more than one Libra ID Domain for a given VASP. We are providing this opportunity to support possible situations when two VASPs merge into one, or when a company wants to provide multiple wallet apps, while having only a single VASP parent account on the chain.
```

Comment:
Probably don't need this in standard, but there will be some natural limitations like account size. In practice I think it will be so rare to have more then one domain per VASP, that we probably don't need to have it as part of the standard?